R/beers-r6.R

```diff
@@ -0,0 +1,153 @@
+Beers_Interpolator <- R6::R6Class("Beers_Interpolator",
+
+  public = list(
+    b1 = NULL,
+    b2 = NULL,
+    bm = NULL,
+    bp = NULL,
+    bf = NULL,
+
+    initialize = function(b1, b2, bm, bp, bf) {
+      self$b1 <- b1
+      self$b2 <- b2
+      self$bm <- bm
+      self$bp <- bp
+      self$bf <- bf
+    },
+
+    interpolate = function(points) {
+      if (length(points) < 6) {
+        stop("Beers interpolation requires at least 6 points")
+
+      } else if (!is.numeric(points)) {
```

Comment:
I tend to make these a series of `if` rather than if/else blocks - simpler to think about